### PR TITLE
New: [Mega Menu] [Infra] Ensure the Mega Menu is A11Y - Core code [ED-9775]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js [hunk 1]
```diff
@@ -325,4 +332,117 @@ export default class NestedTabs extends Base {
...
+			lastItemIsInFocus = this.itemInsideContentContainerHasFocus( 'last' ),
+			activeTabTitleFilter = `.${ this.getActiveClass() }`,
+			$activeTabTitleVisible = this.getVisibleTabTitle( activeTabTitleFilter ),
+			activeTabTitleIndex = parseInt( $activeTabTitleVisible?.getAttribute( 'data-tab' ) ),
+			nextTabTitleFilter = this.getTabTitleFilterSelector( activeTabTitleIndex + 1 ),
+			$nextTabTitleVisible = this.getVisibleTabTitle( nextTabTitleFilter ),
+			pressShiftTabOnFirstFocusableItem = isShiftAndTabPressed && firstItemIsInFocus && !! $activeTabTitleVisible,
+			pressTabOnLastFocusableItem = isOnlyTabPressed && lastItemIsInFocus && !! $nextTabTitleVisible;
+
+		if ( ( pressShiftTabOnFirstFocusableItem ) || isEscapePressed ) {
```
Comment (davseve) [inline modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:386]: Why is `pressShiftTabOnFirstFocusableItem ` wrapped with parentheses?
Can we remove it?

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js [hunk 2]
```diff
@@ -325,4 +332,117 @@ export default class NestedTabs extends Base {
...
+			$nextTabTitleVisible?.focus();
+		}
+	}
+
+	changeFocusFromActiveTabTitleToContentContainer( event ) {
+		const isOnlyTabPressed = 'Tab' === this.getKeyPressed( event ),
+			$focusableItems = this.getFocusableItemsInsideActiveContentContainer(),
+			$firstFocusableItem = $focusableItems[ 0 ],
+			$currentTabTitle = elementorFrontend.elements.window.document.activeElement,
+			currentTabTitleIndex = parseInt( $currentTabTitle?.getAttribute( 'data-tab' ) );
```
Comment (davseve) [inline modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:404]: Is there a situation in which `currentTabTitle` will be undefined?
Why are you using optional chaining?

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js [hunk 3]
```diff
@@ -325,4 +332,117 @@ export default class NestedTabs extends Base {
...
+
+		return settings.classes.active;
+	}
+
+	getVisibleTabTitle( tabTitleFilter ) {
+		const $tabTitle = this.elements.$tabTitles.filter( tabTitleFilter ),
+			isTabTitleDesktopVisible = null !== $tabTitle[ 0 ]?.offsetParent,
+			$visibleTabTitle = isTabTitleDesktopVisible ? $tabTitle[ 0 ] : $tabTitle[ 1 ];
+
+		return $visibleTabTitle;
```
Comment (davseve) [inline modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:347]: The `visibleTabTitle` variable is redundant.
```suggestion
			isTabTitleDesktopVisible = null !== $tabTitle[ 0 ]?.offsetParent;
			return isTabTitleDesktopVisible ? $tabTitle[ 0 ] : $tabTitle[ 1 ];
	}
```

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js [hunk 4]
```diff
@@ -325,4 +332,117 @@ export default class NestedTabs extends Base {
...
+			$focusableItems = this.getFocusableItemsInsideActiveContentContainer(),
+			itemIndex = 'last' === position ? $focusableItems.length - 1 : position;
+
+		return $focusableItems[ itemIndex ] === $currentItem;
+	}
+
+	getFocusableItemsInsideActiveContentContainer() {
+		const settings = this.getSettings();
+
+		return this.$element.find( settings.selectors.activeTabContentContainers ).find( ':focusable' );
```
Comment (davseve) [inline modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:423]: I don't see the definition of `activeTabContentContainers` either here ao in the Pro branch.

**Review by davseve — CHANGES_REQUESTED** [review]
Don't we prefer to place all the accessibility-related functions in a separate handler file?

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js (hunk 4 above)
Comment (davseve) [inline modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:423]: If `getFocusableItemsInsideActiveContentContainer` is set in the mega menu only. what is the reason that this function and all the other functions that are dependent on this one are here and not in the mega menu.

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js (hunk 2 above)
Comment (hein-obox) [reply modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js]: Removed.

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js (hunk 4 above)
Comment (hein-obox) [reply modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:422]: I have moved all functions which will be necessary for the nested tabs accessibility to the nested tabs handler.
We will need this in the next task.

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js (hunk 3 above)
Comment (hein-obox) [reply modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js]: Updated.

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js (hunk 1 above)
Comment (hein-obox) [reply modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js]: Removed

**Review by davseve — APPROVED** [review]

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**